api-authorization/server/authorization-middleware.js

```diff
@@ -0,0 +1,32 @@
+const jwt = require('jsonwebtoken'); // eslint-disable-line
+const ClientError = require('./client-error'); // eslint-disable-line
+
+function authorizationMiddleware(req, res, next) {
+  /* your code here */
+  const tokener = req.headers['x-access-token'];
+  if (!tokener) {
+    throw new ClientError(401, 'authentication required');
+  }
+  jwt.verify(tokener, 'randomtext12345', (err, payload) => {
```

Comment:
When encrypting or decrypting jsonwebtokens, _never_ hard-code the token secret. Instead, pull it from the process environment.